client/rest/src/plugins/metadata/MetadataDb.js

```diff
@@ -78,6 +80,40 @@ class MetadataDb {
 			.toArray()
 			.then(entities => Promise.resolve(this.catapultDb.sanitizer.renameIds(entities)));
 	}
+
+	async binDataByMetalId(metalId) {
+		const compositeHashes = [routeUtils.parseArgument(metal.restoreMetadataHash(metalId), 'compositeHash', 'hash256')];
+		const metadatasByCompositeHash = await this.metadatasByCompositeHash(compositeHashes);
+		const metadataEntry = metal.getMetadataEntryByCompositehash(metadatasByCompositeHash);
+		const chunks = [];
+		let counter = 1;
+
+		const fetchMetadata = async () => {
+			const options = {
+				sortField: 'id', sortDirection: 1, pageSize: 2000, pageNumber: counter
+			};
+			const c = await this.metadata(
+				new Uint8Array(metadataEntry.sourceAddress.buffer),
+				new Uint8Array(metadataEntry.targetAddress.buffer),
+				undefined,
+				metadataEntry.targetId,
+				metadataEntry.metadataType,
+				options
+			);
+			c.data.forEach(e => {
+				chunks.push({
+					key: longToUint64(e.metadataEntry.scopedMetadataKey),
+					value: e.metadataEntry.value.buffer
+				});
+			});
+
+			counter++;
+			if (0 < c.data.length)
+				await fetchMetadata();
```

Comment:
yes, at first it does.
however, if i changed like this, lint is error
`Unexpected `await` inside a loop.eslintno-await-in-loop`
i need `// eslint-disable-next-line no-await-in-loop`
which one is better?
```js
async binDataByMetalId(metalId) {
        const metadatasByCompositeHash = await this.metadatasByCompositeHash([metal.extractCompositeHashFromMetalId(metalId)]);
        if (0 === metadatasByCompositeHash.length)
	        throw Error(`could not get first chunk, it may mistake the metal ID: ${metalId}`);
        const { metadataEntry } = metadatasByCompositeHash[0];
        const chunks = [];
        let counter = 1;
        let moreData = true;
        
        while (moreData) {
	        const options = {
		        sortField: 'id', sortDirection: 1, pageSize: 100, pageNumber: counter
	        };
	        // eslint-disable-next-line no-await-in-loop
	        const metadatas = await this.metadata(
		        metadataEntry.sourceAddress.buffer,
		        metadataEntry.targetAddress.buffer,
		        undefined,
		        metadataEntry.targetId,
		        metadataEntry.metadataType,
		        options
	        );
	        metadatas.data.forEach(e => {
		        chunks.push({
			        key: longToUint64(e.metadataEntry.scopedMetadataKey),
			        value: e.metadataEntry.value.buffer
		        });
	        });
        
	        moreData = 0 < metadatas.data.length;
	        counter++;
        }
        return metal.decode(longToUint64(metadataEntry.scopedMetadataKey), chunks);
        }
```

Reply:
I understood. in my scenario, the futures are dependent. so, i can use `ESLint disable comment` because of according to this article.
if they are independent, i should use Promise.all for improving performance. it means don't use await in loop.
oh i just repeat as your comment😂